README.md

```diff
@@ -12,125 +13,153 @@ npm install @girder/components
+// environment variable is set at build-time
+const apiRoot = process.env.VUE_APP_API_ROOT || 'http://localhost:8080/api/v1';
 
-Many components in this library will require a `RestClient` named `girderRest` through provide/inject.
-The client can be provided through any common ancestor.  For example:
+// Create the axios-based client to be used for all API requests
+const girderRest = new RestClient({
+  apiRoot,
+});
+
+// This is passed to our Vue instance; it will be available in all components
+const GirderProvider = {
+  girderRest,
+};
+export default GirderProvider;
+```
+
+Reference the configuration from your application initialization (typically `src/main.js`):
 
 ```javascript
-import * as girder from '@girder/components';
-// apiRoot should point to your girder instance
-const restClient = new girder.RestClient({ apiRoot: 'http://localhost:8080/api/v1' });
+/* src/main.js */
+import GirderProvider from '@/plugins/girder';
+
+// ...
 
-// Provide "girderRest" to the root
 new Vue({
-  provide: { girderRest },
-  ...
+  provide: GirderProvider,
+  // ...
 }).$mount('#app');
 ```
 
-#### Components
-
-If you're building your own downstream application, you can include individual components from the library. Because these are vuetify components, your consumer application is responsible for creating the `v-app` container.  [Read more](https://vuetifyjs.com/en/layout/pre-defined#all-about-app).  See `demo/main.js` for a more comprehensive example.
-
-Either import the full UMD module:
+### Using Components
 
+Components should be imported by name from `@girder/components/src/components`, as this location will be stable across releases.
+For instance:
 ```javascript
-import Girder, * as girder from '@girder/components';
+import { Upload as GirderUpload } from '@girder/components/src/components';  // Good
+import GirderUpload from '@girder/components/src/components/Upload.vue'; // Unsafe -- may move in future
 ```
 
-Or import specific components from the `src` directory:
+Since Girder web components uses Vuetify, your application must provide
+[`v-app` and `v-content` containers](https://vuetifyjs.com/en/framework/default-markup#all-about-app)
+at some ancestor level.
+
+For example, to create a login / registration widget in `src/App.vue`:
 
 ```html
-<!-- Downstream.vue -->
+<!-- src/App.vue -->
 <template>
-    <girder-upload ... />
+  <v-app>
+    <v-content>
+      <GirderAuthentication register />
+    </v-content>
+  </v-app>
 </template>
 
 <script>
-import { Upload as GirderUpload } from '@girder/components/src/components';
+import { Authentication as GirderAuthentication } from '@girder/components/src/components';
 
 export default {
-    components: { GirderUpload },
-    ...
+  components: {
+    GirderAuthentication,
+  },
+
+  // Injecting is not necessary to use the component,
+  // but is required to access the results of API calls
+  inject: ['girderRest'],
+  created() {
+    this.girderRest.$on('login', this.onLogin);
```

Comment:
I realize this is just a toy example, but I fear people may construe this as a normative pattern for using RestClient.

The `user` property of RestClient is reactive, and is meant to be used as such. I'd suggest we come up with a contrived example of using it reactively, such as 

```pug
div Hello {{ girderRest.user ? girderRest.user.login || "anonymous" }}
```

Reply:
@zachmullen I updated the example.